### PR TITLE
Responsive User Interface Improvements

### PR DESCRIPTION
**Pull request by zbig01**
Have updated the PR 946
 The following changes as requested by @robertdown and @bradymiller have been made
    - use btn-link instead of btn-default in cancel button
    - Made Don't save butons into cancel buttons
    - The cancel button now is visually under emphasized and set apart from other buttons by 20 px
    - Buttons aligned left in All stylesheets except the flat ones ;)
    - Made the all buttons
    - Removed the internal stylesheets to a large extent;))
    - All forms have uniform layout
    - Cleaned up the blank spaces and hopefully formatted code to PSR-2 specs - still doing it on Eclipse
  
  How did I do it:
    - created an external stylesheet oemr-bs3-override.css and wrote some custom Bootstrap rules and changed some default rules
    - imported this into all non-flat stylesheets with @imported
    - This has been done without causing any major disruption of existing Bootstrap 3 styles
    - Changed the box-shadow for buttons on non-light stylesheets to 2px instead of 4px
        
  Do give it a whirl and see if you like it. If not so inclined will upload some eye candy

### CODE REVIEW

**Comment by zbig01** [comment]
Encounter > Clinical

1. Care Plan
care-plan-bootstrap-

2. Clinical Instructions
clinical_instructions-bootstrap-

3. Functional and Cognitive Status
fuctional-cognitive-status-bootstrap-

4. Observation
observation-bootstrap-

5. Review of Systems
ros-bootstrap-

6. Review of Systems - Checks
roschecks-bootstrap-

7. SOAP
soap-bootstrap-

8. Speech Dictation
speech-dictation-bootstrap-

9. Vitals
vitals-bootstrap-

**Comment by zbig01** [comment]
Encounter > Administrative

1. Fee Sheet

fee-sheet-bootstrap-

2. Misc Billing Options

misc-biiling-options-bootstrap-

3. New Encounter
new-encounter-bootstrap-

4. Procedure Order
procedure-order-bootstrap-

**Comment by zbig01** [comment]
Fees

1. Payment
collect-payment-bootstrap-

2. Checkout
checkout-bootstrap-

3.  Billing
billing-bootstrap-

4. Batch Payments - New Payments
new-payment-bootstrap-

5. Search Payment  - 1
search-payment-bootstrap-

6. Search Payment - 2 
search-payment-2-bootstrap-

7. Search Payment - 3
search-payment-3-bootstrap-

8.  ERA Posting
era-posting-bootstrap-

9. EOB Posting - Restyled - New Look 
eob-posting-1-bootstrap-

10. EOB Posting - Invoice Search - For manual entry
eob-posting-2-bootstrap-

11. EOB Posting - Invoice Search - Search Results
eob-posting-3-bootstrap-

12. EOB Posting - Invoice Search - Billing Note
eob-posting-4-bootstrap-

13. EOB Invoice
eob-posting-5-bootstrap-

14. ERA Posting
eob-posting-5-bootstrap-

15. Responsive
era-posting-responsive-bootstrap-

**Comment by bradymiller** [comment]
screenshots look nice!
will look through the code and test it out in the near future.
(One thing I noted is that the Cancel button looks like a disabled button on the light style screenshots above, but let me do some testing first before making any conclusions on that)

**Comment by bradymiller** [comment]
btw, that last screenshot showing 3 screens in responsive mode is very impressive. also noted a modal screenshot too!

**Comment by zbig01** [comment]
Hover and it will wake up


________________________________

**Comment by zbig01** [comment]
Thanks. To the best of my knowledge every one of these pages is fully responsive and looks usable even with 4 locked tabs.


________________________________

File: interface/forms/dictation/new.php [hunk 1]
```diff
@@ -1,42 +1,73 @@
...
+ - * @author  cfapress <cfapress>
+ - * @author  bradymiller <bradymiller@users.sourceforge.net>
+ - * @author  Robert Down <robertdown@live.com>
+ - * @author  Brady Miller <brady.g.miller@gmail.com>
+ - * @author  Brady Miller <brady.g.miller@gmail.com>
+ - * @copyright Copyright (c) 2008 cfapress <cfapress>
+ - * @copyright Copyright (c) 2013 bradymiller <bradymiller@users.sourceforge.net>
+ - * @copyright Copyright (c) 2017 Robert Down <robertdown@live.com>
+ - * @copyright Copyright (c) 2017 Brady Miller <brady.g.miller@gmail.com>
+ - * @copyright Copyright (c) 2016 Brady Miller <brady.g.miller@gmail.com>
```
Comment (robertdown) [inline interface/forms/dictation/new.php]: @bradymiller you've got 3 copyrights on this file :)

File: interface/themes/core.css [hunk 2]
```diff
@@ -7,67 +7,51 @@
  * @copyright Copyright (c) 2017 Robert Down
  * @license https://github.com/openemr/openemr/blob/master/LICENSE General Public License 3
  */
-
```
Comment (robertdown) [inline interface/themes/core.css]: This must stay

File: interface/themes/core.css [hunk 3]
```diff
@@ -7,67 +7,51 @@
...
-@import "themeBuilder.php";
-
 @font-face {
     font-family: "FontAwesome";
     src: url("../../public/assets/font-awesome-4-6-3/fonts/fontawesome-webfont.ttf") format("truetype");
 }
-
```
Comment (robertdown) [inline interface/themes/core.css]: Why remove all these blank lines?

File: interface/themes/core.css [hunk 4]
```diff
@@ -7,67 +7,51 @@
  * @copyright Copyright (c) 2017 Robert Down
  * @license https://github.com/openemr/openemr/blob/master/LICENSE General Public License 3
  */
-
-@import "themeBuilder.php";
```
Comment (robertdown) [inline interface/themes/core.css]: This line has to stay, it allows users to override default theme fonts and font sizes

Comment (robertdown) [inline interface/themes/core.css]: This line has to stay, it allows users to override default theme fonts and font sizes

File: interface/forms/dictation/new.php (hunk 1 above)
Comment (bradymiller) [inline interface/forms/dictation/new.php]: @robertdown ,
only 3 :)
I think this was from the auto generated docblocks(looks like used the github history to produce them) for the PSR2 megacommit to ensure headers at top of all php scripts.

**Comment by robertdown** [comment]
The override style adds too much custom theming. For instance, by default the cancel buttons do not have a border, which is great, because it doesn't add specific rules that require a theme to override. Any classes that we are defining globally should impose as few rules as possible, allowing each theme to set what they want.

For instance, in the Light theme we'd have to go and pull the border, border-radius, shadow, margins, etc.

Also, I think there are some rules that we shouldn't allow themes to change - and button placement is one of them. Since we now have evidence showing buttons should be inline with form inputs, that's where they need to go. Button placement is a core part of how the program flows, allowing a theme to change that introduces lots of concerns for me.

File: interface/themes/core.css (hunk 3 above)
Comment (zbig01) [inline interface/themes/core.css]: Minimizing whitespace

File: interface/billing/billing_report.php [hunk 5]
```diff
@@ -65,1172 +62,1147 @@
...
-    for(i=1;i<=document.getElementById("divnos").value;i++)
-    {
-        var ele = document.getElementById("divid_"+i);
-        if(ele)
-        {
-        set=1;
-        break;
+    <?php //if (function_exists('html_header_show')) html_header_show(); ?>
+    <!--<link rel="stylesheet" href="<?php echo $css_header; ?>" type="text/css">
+    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-datetimepicker-2-5-4/build/jquery.datetimepicker.min.css">-->
```
Comment (bradymiller) [inline interface/billing/billing_report.php]: remove above 3 lines (these are all covered in below setupHeader call)

File: interface/billing/billing_report.php [hunk 6]
```diff
@@ -65,1172 +62,1147 @@
...
+    <?php Header::setupHeader(['datetime-picker','font-awesome']);?>
```
Comment (bradymiller) [inline interface/billing/billing_report.php]: Change to (note font-awesome is included by default in this function):
``````
<?php Header::setupHeader('datetime-picker'); ?>

File: interface/billing/billing_report.php [hunk 7]
```diff
@@ -65,1172 +62,1147 @@
...
+    <style>
+        .subbtn { margin-top:3px; margin-bottom:3px; margin-left:2px; margin-right:2px }
```
Comment (bradymiller) [inline interface/billing/billing_report.php]: seeing a lot of placed in script below where code is being commented out rather than deleted. recommend deleting it instead.

File: interface/billing/edit_payment.php [hunk 8]
```diff
@@ -455,669 +489,695 @@
...
+    <!--<script src="../../library/textformat.js?v=<?php echo $v_js_includes; ?>" type="text/javascript">
+    </script>-->
+    <script src="../../library/dialog.js?v=<?php echo $v_js_includes; ?>" type="text/javascript">
+    </script>
+    <!--<script src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-min-1-7-2/index.js" type="text/javascript">
+    </script>-->
+    <!--<script src="../../library/js/fancybox/jquery.fancybox-1.2.6.js" type="text/javascript">-->
+    </script>
+    <!--<script src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-datetimepicker-2-5-4/build/jquery.datetimepicker.full.min.js" type="text/javascript">
+    </script>-->
```
Comment (bradymiller) [inline interface/billing/edit_payment.php]: remove above commented out code.
also remove dialog.js line since this i automatically included in the setupHeader call

File: interface/themes/oemr-bs3-override.css [hunk 9]
```diff
@@ -0,0 +1,79 @@
+/*override bootstrap 3 for openemr and some new classes all prefixed with oe-*/
```
Comment (bradymiller) [inline interface/themes/oemr-bs3-override.css]: Rec going through this file and categorizing each line into 2 categories:
1. If this is necessary and should never be modified by individual themes. If that is the case, then it should go into core.css.(note we should shoot for ultimate flexibility by each theme developer, so only stuff that is really required should go into this category)
2. Then the rest should be copy/pasted into the themes themselves and then the each theme developer has free reign on this stuff(where the buttons go, how btn-link looks, etc.). As an example, a theme developer may wish to go to bootstrap default legend/fieldset (ie. basically nothing) in their theme along with carnival music playing in the background(you just never know what the future styling fads will be :) ).

**Comment by bradymiller** [comment]
My comment above is very similar to @robertdown's suggestions. We should shoot for as much flexibility as possible for the theme developers. To that end, agree that it makes sense as Robert suggested that codebase should place buttons in line with form inputs as default. However, it seems reasonable that individual themes are ok to override this, if they wish (for example, @zbig01 theme using centered buttons).

File: interface/billing/billing_report.php (hunk 5 above)
Comment (zbig01) [inline interface/billing/billing_report.php]: Will do

File: interface/themes/oemr-bs3-override.css (hunk 9 above)
Comment (zbig01) [inline interface/themes/oemr-bs3-override.css]: The reason this file was created was to be a placeholder for styling that was needed to style the 28 odd pages and minimize use of internal stylesheets.  It is  a start and not the end. I agree that some of this can be consolidated into core.css. 
As always I have tried to keep my changes separate in order to easily identify them and modify as needed.  Some of the styling that was included in here is to ensure that all stylesheets currently in use will display the pages in a consistent way.  Once more pages get styled we will be able to identify what would constitute a 'core' style.
In the development phase the oemr-bs3-override.css is basically a copy and paste operation using  @include and may eventually disappear by being absorbed into core.css or individual stylesheets.
It would certainly help if we include name and ID for elements to help in targeting them for styling without having to modify the code.
A wiki page with the basic styling pertinent to openEMR can be written that would contain styling tips to help convert current non-bootstrap pages and design future pages.

File: interface/forms/soap/templates/general_new.html [hunk 10]
```diff
@@ -1,52 +1,58 @@
 <html>
```
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: still really need to cement out 1 form so gonna focus comments on the soap form

File: interface/forms/soap/templates/general_new.html [hunk 11]
```diff
@@ -1,52 +1,58 @@
...
-                        <div class="col-sm-10">
-                            <textarea name="plan" id="plan" class="form-control" cols="60" rows="6">{$data->get_plan()|escape:'html'}</textarea>
-                        </div>
-                    </div>
-                    <div class="form-group">
-                        <div class="col-sm-offset-2 col-sm-10">
-                            <div class="btn-group" role="group">
+                <form name="soap" method="post" action="{$FORM_ACTION}/interface/forms/soap/save.php" onsubmit="return top.restoreSession()">
+                    <fieldset>
+                            <legend class="">{xl t='Subjective'|escape:'html'}</legend>
```
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: why do all the legend calls have an empty class?

File: interface/forms/soap/templates/general_new.html [hunk 12]
```diff
@@ -1,52 +1,58 @@
...
+                    <fieldset>
+                            <legend class="">{xl t='Plan'|escape:'html'}</legend>
+                                <div class="form-group">
+                                    <div class="col-sm-10 col-sm-offset-1">
+                                        <textarea name="plan" class="form-control" cols="60" rows="6">{$data->get_plan()|escape:'html'}</textarea>
+                                    </div>
+                                </div>
+                    </fieldset>
+                    <?php //can change position of buttons by creating a class 'position-override' and adding rule text-align:center or right as the case may be in individual stylesheets ?>
+                    <div class="form-group clearfix">
```
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: why is clearfix needed here?

File: interface/forms/soap/templates/general_new.html [hunk 13]
```diff
@@ -1,52 +1,58 @@
...
+                        <div class="col-sm-12 text-left position-override">
```
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: The above line doesn't position the button right on wide screen (it should go beneath the input element) and will break in rtl (right to left). Something we haven't discussed is our codebase needs to continue to support rtl. Can't we use the following:
``````
div class="col-sm-10 col-sm-offset-1 position-override"
``````

File: interface/forms/soap/templates/general_new.html [hunk 14]
```diff
@@ -1,52 +1,58 @@
...
+                            <div class="btn-group btn-group-pinch" role="group">
```
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: btn-group-pinch appears to be a theme specific class. this seems fine to have in your theme, but why is this necessary to push in the other themes?

File: interface/forms/soap/templates/general_new.html [hunk 15]
```diff
@@ -1,52 +1,58 @@
...
                                 <button type="submit" class="btn btn-default btn-save" name="Submit">{xl t='Save'|escape:'html'}</button>
-                                <a href="{$DONT_SAVE_LINK}" class="btn btn-link btn-cancel" onclick="top.restoreSession()">{xl t='Cancel'|escape:'html'}</a>
+                                <button type="button" class="btn btn-link btn-cancel btn-separate-left"onclick="top.restoreSession(); location.href='{$DONT_SAVE_LINK}';">{xl t='Cancel'|escape:'html'}</button>
```
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: btn-separate-left appears to be a theme specific class. this seems fine to have in your theme, but why is this necessary to push in the other themes?

File: interface/forms/soap/templates/general_new.html (hunk 11 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: Will remove

File: interface/forms/soap/templates/general_new.html (hunk 12 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: Buttons then lie outside the div. As shown below
With clearfix
clearfix-

Without clearfix
clearfix-without-

Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: Does it impact the view that a user sees(ie. does it cause strange behavior)?

File: interface/forms/soap/templates/general_new.html (hunk 13 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: Look at all the screenshots above in light style, the buttons will tend to shift position from form to form and not be in a consistent location.

Of course not a problem when centerd ;)

File: interface/forms/soap/templates/general_new.html (hunk 14 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: It was designed to give border radius to the last but one button when Cancel button is separted from its bretheren

File: interface/themes/oemr-bs3-override.css (hunk 9 above)
Comment (bradymiller) [inline interface/themes/oemr-bs3-override.css]: I think the issue here is that we don't need to enforce pages to display in a consistent way. Flexibility should trump that.
On my testing would recommend placing the following into core.css:
.form-group
.form-horizontal .control-label

Then the following would go into the individual themes:
legend
fieldset
(note that while playing around I removed the background/background-color entries in each of these elements in light style and actually didn't look that bad albeit the shading is more to my taste; this is an example why this should be left to the theme developers)

And the rest seem to all be theme specific (ie. most don't even need to be included in the other themes, but that can be on a case by case basis; for example, come of the oe* stuff may be needed, but I think the button building and button spacing should be left to the theme; and then the responsibility of enforcing rtl for each theme can be done via demand; so this means the current themes will maintain rtl support while your new themes will require some work if a user/developer wishes to pursue this.

File: interface/forms/soap/templates/general_new.html (hunk 15 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: If removed the cancel button will be right next to the other buttons. in btn-group class.  I thought we wanted the cancel button to be de-emphasized and set apart to prevent accidental clicking a negative/destructive action

File: interface/forms/soap/templates/general_new.html (hunk 13 above)
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: This tests really well in the SOAP form, which I am focusing on. By pulling this to the left rather than lining it up with the input elements, these buttons are actually ending up in the worst place (according to those button articles).

File: interface/forms/soap/templates/general_new.html (hunk 15 above)
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: I don't mean this class should be removed. But that whether this class is supported, should be up to the theme(ie. a them could simply ignore this and not place spacing). would consider the above btn-group-pinch class to be in the same category as this.

File: interface/forms/soap/templates/general_new.html (hunk 12 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: I have seen responsiveness lost in other forms where elements are not confined to their divs

**Comment by MatthewVita** [comment]
@zbig01, I'm very impressed by your screenshots!

File: interface/forms/soap/templates/general_new.html (hunk 12 above)
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: Very interesting that bootstrap has this bug(looks like it's related to floats on a quick google review). Is this a global boostrap form-group bug or is one of our custom classes causing this?

File: interface/forms/soap/templates/general_new.html (hunk 15 above)
Comment (zbig01) [inline interface/forms/soap/templates/general_new.html]: Pl do remember while this can be moved to a custom stylesheet it will have to be included in the code for it to work. on the page.

File: interface/themes/oemr-bs3-override.css [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+
+.form-group {
+	margin-bottom: 5px !Important;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+}
+legend {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+	border-bottom: 2px solid #E5E5E5 !Important;
+	background: #E5E5E5 !Important;
+	padding-left: 10px !Important;
+}
+.form-horizontal .control-label {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+	padding-top: 2px;
+}
+fieldset {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+	background-color: #F2F2F2!Important;
+	margin-bottom: 10px!Important;
+	padding: 0px 0px 15px 0px!Important;
+}
+.btn-link.btn-cancel:focus, .btn-link.btn-cancel:hover {
+	text-decoration: none;
+}
+.btn-link.btn-cancel {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+	border: 1px solid #CCC;
+	border-radius: 3px !Important;
+	color: #8e8e8e !Important;
+}
+.btn-link.btn-cancel.active, .btn-link.btn-cancel.focus,
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+	color: #8e8e8e !Important;
+}
+.btn-link.btn-cancel.active, .btn-link.btn-cancel.focus, 
+.btn-link.btn-cancel:active, .btn-link.btn-cancel:focus,
+.btn-link.btn-cancel:hover, .open>.dropdown-toggle.btn-link.btn-cancel {
+	color: #333 !Important;
+	border-color: #ADADAD;
+	box-shadow: 0px 0px 4px #ADADAD inset;
+}
+.btn-separate-left {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Go ahead and just remove this entire class

File: interface/themes/oemr-bs3-override.css [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+	margin-left: 20px !Important;
+}
+.btn-group>.btn:first-child:not(:last-child):not(.dropdown-toggle) {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme.

File: interface/themes/oemr-bs3-override.css [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+	border-top-right-radius: 3px !Important;
+	border-bottom-right-radius: 3px !Important;
+}
+.btn-group-pinch>.btn:nth-last-child(2):not(.dropdown-toggle) {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Still not sure why we need this specific `btn-group-pinch` class

File: interface/themes/oemr-bs3-override.css [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+	border-top-right-radius: 3px !Important;
+	border-bottom-right-radius: 3px !Important;
+}
+.oe-custom-line{
+   padding-bottom:5px;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Where is this being used?

File: interface/themes/oemr-bs3-override.css [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+}
+.oe-large {
+	display: inline-block;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Should be more abstract, maybe called `.inline` - where is this class being used?

File: interface/themes/oemr-bs3-override.css [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+}
+.oe-small {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Why?

File: interface/themes/oemr-bs3-override.css [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+	display: none;
+}
+.oe-show-hide {
+	display: none;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Use bootstrap's built-in show/hide classes

File: interface/themes/oemr-bs3-override.css [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+}
+.oe-legend-radio{
+    font-size:12px;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Can't inject a font-size here, it will mess up the user's ability to override

File: interface/themes/oemr-bs3-override.css [hunk 30]
```diff
@@ -0,0 +1,79 @@
...
+.oe-large {
+	display: inline-block;
+}
+.oe-small {
+	display: none;
+}
+.oe-show-hide {
+	display: none;
+}
+.oe-legend-radio{
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Put this in your specific theme, not something we can push to every theme

File: interface/themes/oemr-bs3-override.css [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+    font-size:12px; 
+    padding:5px 5px 0px 0px;
+}
+.oe-bold-black {
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Recommend just using `<strong>` if needed something to be bold. And move the color to your theme

File: interface/themes/oemr-bs3-override.css [hunk 32]
```diff
@@ -0,0 +1,79 @@
...
+    font-weight: 700;
+    color: black;
+} 
+.oe-payment-select {
+    width: 80px !important;
+    text-align: right;
+    font-size: 12px;
+}
+.oe-ckbox-label{
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Why this specific class, why not just keep default?

File: interface/themes/oemr-bs3-override.css [hunk 33]
```diff
@@ -0,0 +1,79 @@
...
+    padding-left: 30px;
+    font-weight: 500;
+}
+.oe-file-div{
+    padding:10px 15px;
+}
+.oe-modal-dialog{
+    width:65% !Important;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Where is this being used?

File: interface/themes/oemr-bs3-override.css [hunk 34]
```diff
@@ -0,0 +1,79 @@
...
+    width: 80px !important;
+    text-align: right;
+    font-size: 12px;
+}
+.oe-ckbox-label{
+    padding-left: 30px;
+    font-weight: 500;
+}
+.oe-file-div{
+    padding:10px 15px;
```
Comment (robertdown) [inline interface/themes/oemr-bs3-override.css]: Where is this being used?

**Comment by robertdown** [comment]
Eventually I think it would be good to deprecate the old themes (Except for Light, Radiant, and Tan) as we bring in these new flat styles

File: interface/forms/soap/templates/general_new.html (hunk 15 above)
Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: hi @zbig01 , understood. Definitely leave it here and then the theme styles will have the option of using this or not. and prob would then be good to rename it to something like optional-btn-separate-left.

Comment (bradymiller) [inline interface/forms/soap/templates/general_new.html]: Just saw @robertdown 's review and looks like he is recommending complete removal of this class. Guessing that means there is another way to support this without creating the class, but will be good to get more specifics on this from him.

**Comment by MatthewVita** [comment]
Mentioned this in an email, but I want to make sure it doesn't get missed. The "?" help modals are initially white with no content.

**Comment by MatthewVita** [comment]
Also, something weird is happening on the login screen:
image

**Comment by zbig01** [comment]
Is it http://webagility.com/posts/remove-forced-yellow-input-background-in-chrome
 I am not seeing it at my end.

2017-08-11 23_25_22-openemr login

**Comment by MatthewVita** [comment]
@zbig01 I'm actually talking about the "red bar" on the top. The yellow is just how my Chrome renders inputs due to a special password manager extension I have.

What is the "red bar" for?

**Comment by robertdown** [comment]
Agree with @MatthewVita that the red bar on the main screen should be removed

**Comment by zbig01** [comment]
It is the basis of the color theme. There are a total of 18 color themes, @bradymiller wanted to import one for starters and it happened to be flat red

flat_forest_green
2017-08-12 23_09_57-openemr login-green


flat_tangerine
2017-08-12 23_11_08-openemr login-tangerine


flat_powder_blue

2017-08-12 23_11_54-openemr login-flat-blue

Each of these themes all elements both bootstrap and non-bootstrap are co-ordinated to give a seenless look :))

**Comment by MatthewVita** [comment]
Love the `flat_tangerine` theme!

Perhaps the issue I was having is that the red background seems to blend in with the grey background so it looks like a red bar is floating, unattached to a login form div area. I'm _definately_ not a designer, but maybe adding a darker red background and maybe even border lines will help?

**Comment by zbig01** [comment]
The basis of these flat themes are : 
use a single color as the basic color
achieve variety by adding tints (white) and shades(black) to get subtle variations in color. FYI there are 9 different reds in the color scheme. To get the last red which forms the background  color #F9F3F3 to still look like red and yet be distinguishable  from white and be distinct from the next red further up is a bit of a challenge. 
I tried very hard to find a balance. Because tweaking one color affects the colors up and down the chain.
 I think border lines detract from the flat look. As you may have noticed this login screen is smaller compared to the default login screen which I felt  was too big. I gave it a bit of color to let the user know the color scheme for the rest of the UI.

**Comment by robertdown** [comment]
Input border colors are problematic because they can be used to indicate the validation state of the input. For instance, the red and green themes give colors on the border of text fir,do that could mean "this input is valid" or "this input is invalid." My recommendation is to not colorized the borders, or select a very, _very_ neutral color.

I still think the box feels quite out of place on the login screen.

**Comment by robertdown** [comment]
There are 9 reds in that theme? That seems like a lot, and while perhaps that's the way the theme is supposed to look, it seems like it would be easier to maintain 3 or 4 colors. Plus, at that many colors (even tints/shades) I begin to worry about color fatigue for the user. I've maintained for a long time the over saturated themes of the 90s and early 2000s should be removed from OpenEMR and we should adopt more current design practices. We want the focus or the user to be on the content, not the color; color should be a subtle accent, not the driving force.

**Comment by MatthewVita** [comment]
Thanks for the explanation and rundowns.

My untrained eye sees the red theme as problematic because a) the red box is fading in with the background and the red box above looks like an error. All of the other themes look good to me though.

**Comment by zbig01** [comment]
Heightened the contrast between the body and the login screen and added a border to the login screen.

Procedures > Configuration updated to bootstrap style with new help icon and info icons.  The dialog popup modal has a minor glitch that is preventing the X button from closing the modal  when accessed by clicking the ' Add New Level'  button. It however functions when accessed by clicking the Edot/Add icons.  This modal can be  closed  by clicking the 'cancel' button. To access the info icons Click on ' Add New Level' and hover over the labels in the popup modal. 

Slight redesign of the bootstrap popup modal used for the help file.

Just trying some new features  out. Still a work in progress.

**Comment by bradymiller** [comment]
Continued in this PR:
https://github.com/openemr/openemr/pull/1076
